fdbserver/masterserver.actor.cpp

```diff
@@ -117,8 +117,9 @@ struct MasterData : NonCopyable, ReferenceCounted<MasterData> {
 			forceRecovery = false;
 		}
 		balancer = resolutionBalancer.resolutionBalancing();
-		locality = myInterface.locality.dcId().present() ? std::stoi(myInterface.locality.dcId().get().toString())
-		                                                 : tagLocalityInvalid;
+		locality = (SERVER_KNOBS->ENABLE_VERSION_VECTOR_HA_OPTIMIZATION && myInterface.locality.dcId().present())
```

Comment:
I would suggest not only check dcId is present, but also check it is an integer (in string type)

Reply:
We can ignore that case for now (the knob ENABLE_VERSION_VECTOR_HA_OPTIMIZATION is disabled by default). This patch is to unblock the upgrade (which disables the version vector feature).